draft-iab-use-it-or-lose-it.md

```diff
@@ -638,6 +552,122 @@ This document makes no request of IANA.
+significant routing instability {{RIPE-99}}.
+
+
+## DNS
+
+Ossified DNS code bases and systems resulted in new Resource Record Codes
+(RRCodes) being unusable. A new codepoint would take years of coordination
+between implementations and deployments before it could be relied upon.
+Consequently, overloading use of the TXT record was used to avoid effort and
+delays involved, a method used in the Sender Policy Framework {{?SPF=RFC7208}}
+and other protocols.
+
+It was not until after the standard mechanism for dealing with new RRCodes
+{{?RRTYPE=RFC3597}} was considered widely deployed that new RRCodes can be
+safely created and used.
+
+
+## HTTP
+
+HTTP has a number of very effective extension points in addition to the
+aforementioned header fields.  It also has some examples of extension points
+that are so rarely used that it is possible that they are not at all usable.
+
+Extension points in HTTP that might be unwise to use include the extension point
+on each chunk in the chunked transfer coding {{Section 7.1 of HTTP11}}, the
+ability to use transfer codings other than the chunked coding, and the range
+unit in a range request {{Section 14 of HTTP}}.
+
+
+## IP
+
+The version field in IP was rendered useless when encapsulated over Ethernet,
+requring a new ethertype with IPv6 {{?RFC2464}}, due in part to layer 2 devices
+making version-independent assumptions about the structure of the IPv4 header.
+
+Protocol identifiers or codepoints that are reserved for future use can be
+especially problematic.  Reserving values without attributing semantics to their
+use can result in diverse or conflicting semantics being attributed without any
+hope of interoperability.  An example of this is the "class E" address space in
+IPv4 {{?RFC0988}}, which was originally reserved in {{?RFC0791}} without
+assigning any semantics.
+
+For protocols that can use negotiation to attribute semantics to values, it is
+possible that unused codepoints can be reclaimed for active use, though this
+requires that the negotiation include all protocol participants.  For something
+as fundamental as addressing, negotiation is difficult or even impossible, as
+all nodes on the network path plus potential alternative paths would need to be
+involved.
+
+
+## SNMP
+
+As a counter example, the first version of the Simple Network Management
+Protocol (SNMP) {{?SNMPv1=RFC1157}} defines that unparseable or unauthenticated
+messages are simply discarded without response:
+
+> It then verifies the version number of the SNMP message. If there is a
+  mismatch, it discards the datagram and performs no further actions.
+
+When SNMP versions 2, 2c and 3 came along, older agents did exactly what the
+protocol specifies.  Deployment of new versions was likely successful because
+the handling of newer versions was both clear and simple.
+
+
+## TCP
+
+Extension points in TCP {{?TCP=RFC0793}} have been rendered difficult to use,
+largely due to middlebox interactions.
```

Comment:
Reference: Is it Still Possible to Extend TCP?
Michio Honda, Yoshifumi Nishida, Costin Raiciu, Adam Greenhalgh, Mark Handley and Hideyuki Tokuda, ACM Internet Measurement Conference (IMC), November 2011